src/ucp/core/ucp_context.h

```diff
@@ -643,10 +648,42 @@ int ucp_is_scalable_transport(ucp_context_h context, size_t max_num_eps)
     return (max_num_eps >= (size_t)context->config.est_num_eps);
 }
 
+static UCS_F_ALWAYS_INLINE int
+ucp_context_usage_tracker_enabled(ucp_context_h context)
+{
+    return context->config.ext.dynamic_tl_switch_interval != UCS_TIME_INFINITY;
+}
+
+static UCS_F_ALWAYS_INLINE double
+ucp_tl_iface_latency_with_priority(ucp_context_h context,
+                                   const ucs_linear_func_t *latency,
+                                   int is_prioritized)
+{
+    unsigned num_eps;
+
+    if (!ucp_context_usage_tracker_enabled(context)) {
+        /* No priority EPs exist, use default value */
+        num_eps = context->config.est_num_eps;
+    } else if (is_prioritized) {
+        /* Use small latency for priority EP */
```

Comment:
/* The number of priority endpoints is limited by context->config.ext.max_priority_eps */